packages/components/src/components/common/table/hooks/hooks.tsx

```diff
@@ -119,3 +119,143 @@ export const useMultiSelectRow = <TData,>(table: Table<TData>) => {
 
   return { handleMultiSelectOnRow };
 };
+
+interface KeyHandlerOptions<TData> {
+  multiSelect?: boolean;
+  reorder?: { updateOrder?: (moveIndexes: number[], toIndex: number) => void; getRowId?: (row: TData) => string };
+  lazyLoadChildren?: (row: Row<TData>) => void;
+}
+
+interface TableKeyboardHandlerProps<TData> {
+  table: Table<TData>;
+  data: Array<TData>;
+  options?: KeyHandlerOptions<TData>;
+}
+
+export const useTableKeyHandler = <TData,>({ table, data, options }: TableKeyboardHandlerProps<TData>) => {
+  const [rootIndex, setRootIndex] = React.useState<number | undefined>(undefined);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLTableElement>, onEnterAction?: (row: Row<TData>) => void) => {
+    event.stopPropagation();
+
+    switch (event.key) {
+      case 'ArrowUp':
+      case 'ArrowDown': {
+        handleArrowKeyUpDown(event);
+        break;
+      }
+      case 'ArrowRight':
+        toggleExpand(table.getSelectedRowModel().flatRows, true, options?.lazyLoadChildren);
+        break;
+      case 'ArrowLeft':
+        toggleExpand(table.getSelectedRowModel().flatRows, false, options?.lazyLoadChildren);
+        break;
+      case 'Tab':
+        table.resetRowSelection();
+        break;
+      case 'Enter':
+        handleEnterKey(table.getSelectedRowModel().flatRows, onEnterAction);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleArrowKeyUpDown = (event: React.KeyboardEvent<HTMLTableElement>) => {
+    const { multiSelect = false, reorder } = options || {};
+    const allRows = table.getRowModel().rows;
+    const selectedRows = table.getSelectedRowModel().flatRows;
+    const direction = event.key === 'ArrowUp' ? -1 : 1;
```

Comment:
You now check this key twice. I think it is better if you do it like the ArrowRight/Left way. With a direction param